### PR TITLE
Update REPL to more efficiently auto-complete ifql files and dirs

### PR DESCRIPTION
**Pull request by nathanielc**
With this change the REPL is fast, there is no perceivable lag. But the IFQL process is idling at 15% CPU on my box. We need a better solution. Maybe we restrict auto-completion to files in the current dir instead of recursively traversing all files?

### CODE REVIEW

File: repl/filewatcher.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	ticker := time.NewTicker(500 * time.Millisecond)
+	for range ticker.C {
+		w.update()
+	}
+}
+
+func (w *filewatcher) update() {
+	list := make([]string, 0, 10)
+
+	filepath.Walk(w.root, func(path string, info os.FileInfo, err error) error {
```
Comment (aanthony1243) [inline repl/filewatcher.go]: yeah, maybe we can just glob *.ifql on the CWD and forget about recursing.  I did not think through carefully that this search would be performed at every invocation of the auto-complete.

File: repl/filewatcher.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	w := &filewatcher{
+		root: root,
+	}
+	w.update()
+	go w.run()
+	return w
+}
+
+func (w *filewatcher) run() {
+	ticker := time.NewTicker(500 * time.Millisecond)
```
Comment (aanthony1243) [inline repl/filewatcher.go]: might get away with every 1s too.  This only updates when there's a new file in the dir.

**Comment by nathanielc** [comment]
OK, will do. It will be easy enough to auto-complete dirs as they type them too

**Comment by nathanielc** [comment]
@aanthony1243 I changed it so there is no background process, but its still has no perceivable lag.

Basically we only read the current dir or whatever dir the user has entered.

A nice addition with this change is that now this will autocomplete
```
> @../something.ifql
```

We can work on improving this more later but for now it should be plenty fast and still quite handy.

File: repl/repl.go [hunk 3]
```diff
@@ -221,25 +231,24 @@ func (r *REPL) doQuery(spec *query.Spec) error {
 	return nil
 }
 
-func getIfqlFiles(rootpath string) []string {
-
-	list := make([]string, 0, 10)
+func getIfqlFiles(path string) ([]string, error) {
+	dir, name := filepath.Split(path)
```
Comment (aanthony1243) [inline repl/repl.go]: small nit:  it seems that you don't need to split the path here, wouldn't   path + "*.ifql"   get you the same result?

Comment (nathanielc) [reply repl/repl.go]: ah yes it would. I'll fix that

**Pull request merged**